File: C++/generateTriangle.cpp
class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        vector<vector<int> > result;
        
        for (size_t i = 0; i < numRows; i++) {
            vector<int> row;
            for (size_t j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    row.push_back(1);
                }
                else {
                    row.push_back(result[i-1][j-1] + result[i-1][j]);
                }
            }
            result.push_back(row);
        }
        
        return result;
    }
};

File: Java/generateTriangle.java
public class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ret = new ArrayList<List<Integer>>();

        for (int i = 0; i < numRows; i++) {
            List<Integer> temp = new ArrayList<Integer>();
            for (int j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    temp.add(1);
                }
                else {
                    temp.add(ret.get(i-1).get(j-1) + ret.get(i-1).get(j));
                }
            }
            ret.add(temp);
        }

        return ret;
    }
}
